### ISSUE TITLE
Compiler crashes silently with certain usages of `usingnamespace`

### ISSUE THREAD

**Issue by expikr** [ziglang/zig#18001]
### Zig Version

* 0.11.0
* 0.12.0-dev.1645+7b99189f1

### Steps to Reproduce and Observed Behavior

The following code crashes the compiler without any output whatsoever:

```zig
pub fn main() void {
    const TestThing = opaque{};
    const a = AbstractSingletonFactory(TestThing);
    a.print();
    a.print();
    const b = AbstractSingletonFactory(TestThing);
    b.print();
    a.print();
}


inline fn AbstractSingletonFactory(comptime Thing: type) *opaque{
    usingnamespace Thing;
    var state:usize = 0;
    pub fn print(_:*@This()) void { 
        @import("std").debug.print("hi {}\n",.{state});
        state += 1;
    }
} {
    return undefined;
}
```

On the other hand, removing `usingnamespace` works:

```zig
pub fn main() void {
    const a = AbstractSingletonFactory();
    a.print();
    a.print();
    const b = AbstractSingletonFactory();
    b.print();
    a.print();
}


inline fn AbstractSingletonFactory() *opaque{
    var state:usize = 0;
    pub fn print(_:*@This()) void { 
        @import("std").debug.print("hi {}\n",.{state});
        state += 1;
    }
} {
    return undefined;
}
```

### Expected Behavior

Should print out what's wrong with the code.

**Comment by rootbeer** [comment]
Reproduced with a backtrace:
```
$ zig build-obj --debug-log all t.zig
thread 2677092 panic: reached unreachable code
Analyzing t.zig: t.zig:AbstractSingletonFactory__opaque_3417.usingnamespace_0
    > %52 = closure_get(%51)
      %53 = break_inline(%50, %52)
    For full context, use the command
      zig ast-check -t t.zig


/home/pat/projects/ziglang/zig/lib/std/debug.zig:342:14: 0x614a9bc in assert (zig)
    if (!ok) unreachable; // assertion failure
             ^
/home/pat/projects/ziglang/zig/src/Sema.zig:16688:15: 0x6b30f4a in zirClosureGet (zig)
        assert(scope != .none);
              ^
/home/pat/projects/ziglang/zig/src/Sema.zig:1030:68: 0x67c374f in analyzeBodyInner (zig)
            .closure_get                  => try sema.zirClosureGet(block, inst),
                                                                   ^
/home/pat/projects/ziglang/zig/src/Sema.zig:933:45: 0x6536d44 in analyzeBodyBreak (zig)
    const break_inst = sema.analyzeBodyInner(block, body) catch |err| switch (err) {
                                            ^
/home/pat/projects/ziglang/zig/src/Module.zig:3668:50: 0x6534221 in semaDecl (zig)
    const result_ref = (try sema.analyzeBodyBreak(&block_scope, @ptrCast(body))).?.operand;
                                                 ^
/home/pat/projects/ziglang/zig/src/Module.zig:3242:32: 0x62a067d in ensureDeclAnalyzed (zig)
        break :blk mod.semaDecl(decl_index) catch |err| switch (err) {
                               ^
/home/pat/projects/ziglang/zig/src/Compilation.zig:3820:38: 0x65188cb in processOneJob (zig)
            module.ensureDeclAnalyzed(decl_index) catch |err| switch (err) {
                                     ^
/home/pat/projects/ziglang/zig/src/Compilation.zig:3691:30: 0x62e25d7 in performAllTheWork (zig)
            try processOneJob(comp, work_item, main_progress_node);
                             ^
/home/pat/projects/ziglang/zig/src/Compilation.zig:2500:31: 0x62dde14 in update (zig)
    try comp.performAllTheWork(main_progress_node);
                              ^
/home/pat/projects/ziglang/zig/src/main.zig:4279:24: 0x630dc82 in updateModule (zig)
        try comp.update(main_progress_node);
                       ^
/home/pat/projects/ziglang/zig/src/main.zig:3680:17: 0x632dfa1 in buildOutputType (zig)
    updateModule(comp) catch |err| switch (err) {
                ^
/home/pat/projects/ziglang/zig/src/main.zig:283:31: 0x614cd8f in mainArgs (zig)
        return buildOutputType(gpa, arena, args, .{ .build = .Obj });
                              ^
/home/pat/projects/ziglang/zig/src/main.zig:223:20: 0x6149e95 in main (zig)
    return mainArgs(gpa, arena, args);
                   ^
/home/pat/projects/ziglang/zig/lib/std/start.zig:585:37: 0x6149916 in main (zig)
            const result = root.main() catch |err| {
                                    ^
../sysdeps/nptl/libc_start_call_main.h:58:16: 0x7f75ec8321c9 in __libc_start_call_main (../sysdeps/x86/libc-start.c)
../csu/libc-start.c:360:3: 0x7f75ec832284 in __libc_start_main_impl (../sysdeps/x86/libc-start.c)
???:?:?: 0x6149560 in ??? (???)
???:?:?: 0x0 in ??? (???)
Aborted
```

I believe its the assert at the end of this block in `zirClosureGet`:
```
    // Note: The target closure must be in this scope list.
    // If it's not here, the zir is invalid, or the list is broken.
    const capture_ty = while (true) {
        // Note: We don't need to add a dependency here, because
        // decls always depend on their lexical parents.
        const key: CaptureScope.Key = .{
            .zir_index = inst_data.inst,
            .index = scope,
        };
        if (mod.comptime_capture_scopes.get(key)) |val|
            return Air.internedToRef(val);
        if (mod.runtime_capture_scopes.get(key)) |ty|
            break ty;
        scope = scope.parent(mod);
        assert(scope != .none);
    };
```

**Comment by xdBronch** [comment]
this compiles and runs using `0.12.0-dev.3439+31a7f22b8`

**Comment by Vexu** [comment]
Fixed by #19190
